package/MDAnalysis/coordinates/GMS.py

```diff
@@ -126,45 +124,49 @@ def n_atoms(self):
             return self._n_atoms
 
     def _read_out_natoms(self):
-        self._reopen()
-        # this assumes that this is only called once at startup and that the
-        # filestream is already open
-        for line in self.outfile:
-            m = re.match(r'\s*TOTAL NUMBER OF ATOMS\s*=\s*([0-9]+)\s*',line)
-            if m is None:
-                continue
-            self.close()
-            return int(m.group(1))
+        with util.anyopen(self.filename, 'rt') as out:
+            for line in out:
+                m = re.match(r'\s*TOTAL NUMBER OF ATOMS\s*=\s*([0-9]+)\s*',line)
+                if not m is None:
+                    res = int(m.group(1))
+                    break
 
-        self.close()
-        raise EOFError
+        return res
 
     @property
     def n_frames(self):
         if self._n_frames is not None:   # return cached value
             return self._n_frames
         try:
-            self._n_frames = self._read_out_n_frames(self.filename)
+            self._n_frames = self._read_out_n_frames()
         except IOError:
             return 0
         else:
             return self._n_frames
 
-    def _read_out_n_frames(self, filename):
-        self._reopen()
-        counter = 0
+    def _read_out_n_frames(self):
         if self.runtyp == 'optimize':
-            for line in self.outfile:
-                if re.match(r'^.NSERCH=.*', line):
-                    counter += 1
+            trigger = re.compile(r'^.NSERCH=.*')
         elif self.runtyp == 'surface':
-            for line in self.outfile:
-                if re.match(r'^.COORD 1=.*', line):
+            trigger = re.compile(r'^.COORD 1=.*')
+
+        counter = 0
+        self._offsets = offsets = []
+        with util.anyopen(self.filename, 'rt') as out:
+            line = True
+            while not line == '':  # while not EOF
```

Comment:
Simplify to 
```python
line = "read me"
while line.strip():
```
maybe?

I am actually not sure that your current code ever triggers because all lines will also contain a newline and will never be empty (unless the last line in the file happens to be an empty line without newline but then there should be a cleaner way to exit).